src/Microsoft.Framework.BufferEntryCollection.Internal/BufferEntryCollection.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.Framework.Internal
+{
+    /// <summary>
+    /// An ordered collection of strings with an enumerator that iterates over it.
+    /// </summary>
+    internal class BufferEntryCollection : IEnumerable<string>
+    {
+        // Specifies the maximum size we'll allow for direct conversion from
+        // char arrays to string.
+        private const int MaxCharToStringLength = 1024;
+        private readonly List<object> _buffer = new List<object>();
+        private bool _isModified = false;
```

Comment:
If the user wrote something to it and cleared, we want IsModified to return true. So yes, we never want it to flip back.